### PR TITLE
Vault Timeout Service Tests & Bug Fix

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

While working on a different ticket I started working on tests for VaultSettingsService and noticed a bug while writing the tests. The other ticket may not be completed as quickly so I decided to make a PR for these items instead of waiting for the other work.

Helps with PM-4996

## Code changes

- **libs/common/src/services/vault-timeout/vault-timeout.service.spec.ts:** Add a lot of tests to cover the service.
- **libs/common/src/services/vault-timeout/vault-timeout.service.ts:** Fix a bug where we would be getting the vault timeout actions for the current active user vs getting them for the user actively locking.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/common/src/services/vault-timeout/vault-timeout.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+        });
+
+        await vaultTimeoutService.checkVaultTimeout();
+
+        expectNoAction("1");
+      }
+    );
+
+    it.each([undefined, null])(
+      "should not log out or lock a user who has %s lastActive value",
```
Comment (MGibson1) [inline libs/common/src/services/vault-timeout/vault-timeout.service.spec.ts:202]: **?**: does jest correctly name these tests? null and undefined stringify weird sometimes

File: libs/common/src/services/vault-timeout/vault-timeout.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+      // Act
+      await vaultTimeoutService.checkVaultTimeout();
+
+      // Assert
+      expectUserToHaveLocked("1");
+      expectNoAction("2");
+    });
+
+    it("should run action on an account that hasn't been active for greater than 1 minute and has a vault timeout for 1 minutes", async () => {
```
Comment (MGibson1) [inline libs/common/src/services/vault-timeout/vault-timeout.service.spec.ts:245]: **nit**: I prefer single assert tests, which would make this test a describe block with four tests in it

**Review by MGibson1 — APPROVED** [review]

File: libs/common/src/services/vault-timeout/vault-timeout.service.spec.ts (hunk 1 above)
Comment (justindbaur) [reply libs/common/src/services/vault-timeout/vault-timeout.service.spec.ts:202]: It does! 
Screenshot 2023-11-29 at 9 10 33 AM

File: libs/common/src/services/vault-timeout/vault-timeout.service.ts [hunk 3]
```diff
@@ -71,7 +71,7 @@ export class VaultTimeoutService implements VaultTimeoutServiceAbstraction {
     }
 
     const availableActions = await firstValueFrom(
-      this.vaultTimeoutSettingsService.availableVaultTimeoutActions$()
+      this.vaultTimeoutSettingsService.availableVaultTimeoutActions$(userId)
```
[the remark this answers is not in the document]
Comment (justindbaur) [reply libs/common/src/services/vault-timeout/vault-timeout.service.ts:74]: Yeah... I did try and see if I could make it non-optional but other uses of the method utilize it's optional-ness...

**Pull request merged**